include/type_traits/traits_shared.hpp

```diff
@@ -64,12 +65,58 @@ struct ContainersSharedTraits
   static constexpr int rank = _rank;
 };
 
+template<
+  typename Ordinal,
+  typename Size = Ordinal
+>
+struct OrdinalTrait
+{
+  using ordinal_type = Ordinal;
+  using size_type    = Size;
+};
+
+template<
+  typename Scalar,
+  typename ScalarRef = Scalar&,
+  typename ScalarConstRef = typename std::add_const<ScalarRef>::type
+>
+struct ScalarTrait
+{
+  using scalar_type    = Scalar;
+  using reference_type = ScalarRef;
+  using const_reference_type = ScalarConstRef;
+};
+
+template<bool _is_static>
+struct AllocTrait
```

Comment:
this name is a bit unclear, can we make it more precise? maybe `StaticAllocationTrait` ?

Reply:
In [_traits_shared.hpp:97-98_](https://github.com/Pressio/pressio/blob/tests_302/include/type_traits/traits_shared.hpp#L97-L98) we have `StaticAllocTrait` and `DynamicAllocTrait` defined, which name the allocation directly and are to be used unless it depends on the type like in [`EigenMatrixAllocTrait<T>`](https://github.com/Pressio/pressio/blob/tests_302/include/type_traits/traits_tpl.hpp#L246-L250)